src/handlers/ExecuteViewSubmitHandler.ts

```diff
@@ -421,4 +429,180 @@ export class ExecuteViewSubmitHandler {
             await sendNotification(this.read, this.modify, user, room, { message });
         }
     }
-} 
\ No newline at end of file
+
+    private async handleLLMConfigurationSubmit(user: any, view: any): Promise<IUIKitResponse> {
+        let room: IRoom | undefined;
+
+        try {
+            const roomInteractionStorage = new RoomInteractionStorage(
+                this.persistence,
+                this.read.getPersistenceReader(),
+                user.id,
+            );
+            const roomId = await roomInteractionStorage.getInteractionRoomId();
+            room = roomId ? await this.read.getRoomReader().getById(roomId) : undefined;
+
+            const currentLanguage = await getUserPreferredLanguage(
+                this.read.getPersistenceReader(),
+                this.persistence,
+                user.id,
+            );
+
+            const llmUsagePreferenceValue = this.getFormValue(view.state, LLMConfigurationModalEnum.LLM_USAGE_PREFERENCE_DROPDOWN_ACTION_ID);
+            const llmProviderValue = this.getFormValue(view.state, LLMConfigurationModalEnum.LLM_PROVIDER_DROPDOWN_ACTION_ID);
+            const selfHostedUrlValue = this.getFormValue(view.state, LLMConfigurationModalEnum.SELF_HOSTED_URL_ACTION_ID);
+            const openaiApiKeyValue = this.getFormValue(view.state, LLMConfigurationModalEnum.OPENAI_API_KEY_ACTION_ID);
+            const geminiApiKeyValue = this.getFormValue(view.state, LLMConfigurationModalEnum.GEMINI_API_KEY_ACTION_ID);
+            const groqApiKeyValue = this.getFormValue(view.state, LLMConfigurationModalEnum.GROQ_API_KEY_ACTION_ID);
+
+            if (llmUsagePreferenceValue === LLMUsagePreferenceEnum.Personal) {
+                const validationError = this.validatePersonalLLMConfig(
+                    llmProviderValue,
+                    selfHostedUrlValue,
+                    openaiApiKeyValue,
+                    geminiApiKeyValue,
+                    groqApiKeyValue,
+                    currentLanguage
+                );
+
+                if (validationError) {
+                    if (room) {
+                        await sendNotification(this.read, this.modify, user, room, {
+                            message: validationError
+                        });
+                    }
+
+                    return this.context.getInteractionResponder().errorResponse();
+                }
+            }
+
+            const userPreferenceStorage = new UserPreferenceStorage(
+                this.persistence,
+                this.read.getPersistenceReader(),
+                user.id,
+            );
+            const existingPreference = await userPreferenceStorage.getUserPreference();
+
+            const currentLLMConfig = existingPreference.llmConfiguration || {} as any;
+            const updatedPreference = {
+                ...existingPreference,
+                llmConfiguration: {
```

Comment:
The way thise object it managed looks really bad can you try to find better way to manage this logic ?